### DISCUSSION TITLE
Proposed Roadmap for Q4 2021

### DISCUSSION THREAD

**Discussion by kostko** [oasisprotocol/community-forum#29] [Mainnet]
## Overview

The Oasis team is hard at work building new features for the Oasis Network and driving growth and adoption of the network as a whole. Here's a quick a summary of what we're focusing on this quarter:
* Upgrading the [Oasis Wallet - Browser Extension] to support interacting with ParaTimes.
* Launching an initial version of the [Cipher ParaTime] on Mainnet.
* Finalizing the [EVM module for the Oasis SDK] and a new [Web3 gateway] to go with it in order to enable existing Ethereum applications to work inside ParaTimes.
* Launching an EVM-compatible ParaTime dedicated to DeFi on Mainnet.
* Integrating the cross-chain [Wormhole bridge] with the EVM ParaTime.
* Enable community developers to launch DEX (decentralized exchange) and Lending protocols on the Emerald ParaTime.
* Finishing developer documentation and high-level utilities for interacting with the WebAssembly smart contracts module to improve user experience.

For next quarter (Q1 2022), we propose to focus on:
* Launching the [WebAssembly smart contracts module] on the Cipher Mainnet with confidential DeFi apps.
* Launching more DeFi apps on the EVM ParaTime.
* General availability of data tokenization in the Parcel ParaTime.

We are also planning on proposing a network upgrade that will increase performance of both the consensus layer and both confidential and non-confidential ParaTimes.

[Oasis Wallet - Browser Extension]: https://github.com/oasisprotocol/oasis-wallet-ext
[Cipher ParaTime]: https://github.com/oasisprotocol/cipher-paratime
[EVM module for the Oasis SDK]: https://github.com/oasisprotocol/oasis-sdk/tree/main/runtime-sdk/modules/evm
[Web3 gateway]: https://github.com/starfishlabs/oasis-evm-web3-gateway
[Wormhole bridge]: https://github.com/certusone/wormhole
[WebAssembly smart contracts module]: https://github.com/oasisprotocol/oasis-sdk/tree/main/runtime-sdk/modules/contracts
[Parcel]: https://medium.com/oasis-protocol-project/introducing-parcel-beta-34d729553d8e

## Technology Roadmap

With the expansion of the Oasis SDK to cover WebAssembly and EVM-based smart contracts, the technology focus in this and next quarters will be on improving the developer and user experience when interacting with ParaTimes, expand interoperability between ParaTimes and with other networks, further improvements to the architecture to increase performance of the various components, and supporting the novel confidential use cases with strong security and privacy guarantees.

### EVM-compatible ParaTime

In order to support existing Ethereum applications to work within the Oasis ecosystem, we will be launching a new ParaTime built around the Oasis SDK and the already developed [EVM module] to enable ROSE use inside EVM smart contracts.

While previously compatibility with Ethereum was provided by the [Oasis-Eth ParaTime] developed by an external team, this new initiative is independent and is entirely based on the Oasis SDK which makes it is more easily interoperable with the wider Oasis and ROSE ecosystem. The team behind the Oasis-Eth ParaTime is meanwhile re-focusing their efforts on EVM-based confidential compute and building eWASM developer tools that will benefit the Cipher ParaTime.

The new EVM-compatible ParaTime will enable easy porting of existing audited codebases while at the same time opening a path for EVM smart contracts to incrementally adopt confidentiality features provided by the Cipher ParaTime. And since ROSE will be the native token in both ParaTimes this will make interoperability easier and also greatly expand its utility.

[EVM module]: https://github.com/oasisprotocol/oasis-sdk/tree/main/runtime-sdk/modules/evm
[Oasis-Eth ParaTime]: https://www.oasiseth.org/

### Cipher ParaTime

An initial version of the Cipher ParaTime will launch on Mainnet first, followed by upgrades which will enable WebAssembly-based smart contracts that are already deployed on Testnet and finally confidentiality features. Since the base set of features for Cipher is complete, we will be focusing on testing and improving developer documentation and tooling to make the development of confidential smart contracts a pleasant and safe experience.

## Network Adoption

We often talk about the technology we're building to support the growth of the Oasis Network. In this section, we want to update you on other equally-important pieces of work that are impacting how developers adopt the technology and improvements coming to the Oasis Network.

### DeFi

With the upcoming launch of the Cipher and EVM-compatible ParaTimes on Mainnet, the Oasis Network will have developer friendly support for both confidential and EVM-based smart contracts. This will enable the numerous DeFi partnerships already working on their integrations to more easily and incrementally leverage the unique privacy properties of the Oasis Network. With most of the technical work completed, our goal during the coming quarters will be largely focused on expanding the pool of active DeFi projects on the Oasis Network by improving developer friendliness and user experience.

**Privacy-preserving DEX**

DeFi applications have gained a lot of momentum in Ethereum over the past year. That said, it faces major problems (e.g. low through-put, high gas fees and front-running / MEV problems, etc.) that prevent DeFi from going mainstream via the Ethereum network. A private DEX using the Cipher ParaTime with support for confidential compute is best positioned to solve those problems. The Oasis Foundation is collaborating with a DEX team to co-build a privacy-preserving DEX, which is expected to launch in 2022.

### $1.5M in Ecosystem Grants

Because of the progress the team has made to make the Oasis Network ready and given the excellent performance of the upgrades, the Oasis community has voted to commit up to $1.5M in Grants to support the development of ideas that either improve the Oasis Network or leverage it to build new use cases. [Apply here].

### New partnerships

Partnerships have been a critical part of the Oasis Network; from node operators to DeFi primitives and Dapps, the Oasis Network will continue to attract the newest and most popular players in Blockchain. We will constantly release updates on this subject, with a steady cadence of new announcements coming out each month.

### Looking Forward

This roadmap is just the beginning. We have a lot more in store for the coming quarters, so please join our [Telegram] or follow us on [Twitter] to stay up to date on the latest news and activities!

[Apply here]: https://oasisfoundation.typeform.com/to/HtYql2aN
[Telegram]: https://t.me/oasisprotocolcommunity
[Twitter]: https://twitter.com/OasisProtocol

**Comment by ATTILASAMUELL** [comment]
Very good.
